nomad/job_endpoint.go

```diff
@@ -175,11 +176,17 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 		}
 	}
 
+	if ok, err := allowedRegistration(aclObj, j.srv.State()); !ok || err != nil {
+		j.logger.Warn("job registration for non-management ACL rejected")
+		return structs.ErrPermissionDenied
```

Comment:
From a sidebar discussion: we'd like this error to result in a 403 at the API client but have richer information about _why_ the feature is disabled that we usually don't return.

Reply:
Done